[first phase] Functions user guide revamp (#1196)
Macrometacorp/docs: docs/photoniq/functions/index.md

---
sidebar_position: 1
title: Functions
---

> PhotonIQ functions is an edge-computing service that allows you to run your applications smoothly without the hassle of managing infrastructure, server and resources. Utilizing the highly distributed [Global Data Mesh](https://www.macrometa.com/platform/global-data-mesh) to deploy and execute functions from the closest point of presence (PoP) to customers, it ensures low-latency and high-performance. PhotonIQ functions allow you to implement your system's logic using event-driven, ready-to-use blocks of code called "functions".

Reviewer: ```suggestion
PhotonIQ Functions is an edge-computing service that allows you to run your applications smoothly without the hassle of managing infrastructure, server and resources. Utilizing the highly distributed [Global Data Mesh](https://www.macrometa.com/platform/global-data-mesh) to deploy and execute functions from the closest point of presence (PoP) to customers, it ensures low-latency and high-performance. PhotonIQ Functions allow you to implement your system's logic using event-driven, ready-to-use blocks of code called "functions".
```

  />
 </grid> 

## How PhotonIQ Functions work

> PhotonIQ Functions use an event-driven model to trigger and deploy lightweight, high-performance functions as [WebAssembly](https://webassembly.org/) modules in response to HTTP requests. This model ensures efficient resource consumption, as the Functions service only utilizes resources when needed. Additionally, the stateless nature of the service allows it to handle multiple requests independently and in parallel, thereby enhancing performance.

Reviewer: The event-driven mention again.

```suggestion
PhotonIQ Functions uses a serverless model to trigger and deploy lightweight, high-performance functions as [WebAssembly](https://webassembly.org/) modules in response to HTTP requests. This model ensures efficient resource consumption, as the Functions service only utilizes resources when needed. Additionally, the stateless nature of the service allows it to handle multiple requests independently and in parallel, thereby enhancing performance.
```
